dleq/go_dleq.go

```diff
@@ -0,0 +1,76 @@
+package dleq
+
+import (
+	csecp256k1 "github.com/athanorlabs/atomic-swap/crypto/secp256k1"
+
+	dleq "github.com/noot/go-dleq"
+	"github.com/noot/go-dleq/ed25519"
+	"github.com/noot/go-dleq/secp256k1"
+
+	dsecp256k1 "github.com/decred/dcrd/dcrec/secp256k1/v4"
+)
+
+// DefaultDLEq is the default DLEq prover.
+type DefaultDLEq = GoDLEq
+
+// GoDLEq is a wrapper around the go-dleq library prover and verifier.
+type GoDLEq struct{}
+
+var (
+	curveA = secp256k1.NewCurve()
+	curveB = ed25519.NewCurve()
+)
+
+// Prove generates a secret and a corresponding proof that it has a value
+// on the secp256k1 and ed25519 curves.
+func (d *GoDLEq) Prove() (*Proof, error) {
```

Comment:
The name doesn't seem to reflect what the function is doing. What would be more self documenting?  `GenSecretAndProof`? For the comment, I don't understand what it means for a secret to have a value on 2 curves. That sounds like a point, but I'm assuming the secret is a scalar that only forms a point on the curve when it is multiplied by the curve's generator (`G`).

Reply:
yes it means it has a corresponding point on both curves when multiplied by the curves's generators. will update